scripts/gen_gitlab_config.py

```diff
@@ -272,47 +272,51 @@ def gen_build_base_venvs() -> None:
     PIP_CACHE_DIR: '${{CI_PROJECT_DIR}}/.cache/pip'
     SCCACHE_DIR: '${{CI_PROJECT_DIR}}/.cache/sccache'
     DD_FAST_BUILD: '1'
+    DD_CMAKE_INCREMENTAL_BUILD: '1'
+    DD_SETUP_CACHE_DOWNLOADS: '1'
+    EXT_CACHE_VENV: '${{CI_PROJECT_DIR}}/.cache/ext_cache_venv'
   rules:
     - if: '$CI_COMMIT_REF_NAME == "main"'
       variables:
         DD_FAST_BUILD: '0'
     - when: always
   script: |
     set -e -o pipefail
-    if [ ! -f cache_used.txt ];
-    then
-      echo "No cache found, building native extensions and base venv"
-      apt update && apt install -y sccache
-      pip install riot==0.20.1
-      riot -P -v generate --python=$PYTHON_VERSION
-      echo "Running smoke tests"
-      riot -v run -s --python=$PYTHON_VERSION smoke_test
-      touch cache_used.txt
+    apt update && apt install -y sccache
+    pip install riot==0.20.1
```

Comment:
Perhaps we can include riot in the cached venv?

Reply:
This is only a few seconds over a 2m run. Maybe worth considering together with dropping sccache.